test/test_classifier.rb

```diff
@@ -26,17 +27,18 @@ def test_restricted_classify
     Classifier.train! db, "Ruby", fixture("Ruby/foo.rb")
     Classifier.train! db, "Objective-C", fixture("Objective-C/Foo.h")
     Classifier.train! db, "Objective-C", fixture("Objective-C/Foo.m")
+    Classifier.finalize_train! db
 
     results = Classifier.classify(db, fixture("Objective-C/hello.m"), ["Objective-C"])
     assert_equal "Objective-C", results.first[0]
 
     results = Classifier.classify(db, fixture("Objective-C/hello.m"), ["Ruby"])
-    assert_equal "Ruby", results.first[0]
+    assert results.empty?
```

Comment:
We are currently returning all languages with a score. If there is no matched token at all, all languages are returned with the same score (some arbitrarily low number) and their order is undefined.

With this PR, I'm returning only languages where there is at least one matched token. I think this behaviour is desirable, since it's better to say "no match" rather than "some random match". However, it can certainly break backwards compatibility. If we want to keep the current behavior, we could return all languages, arbitrarily ordered, with score 0.